packages/back-end/src/queryRunners/ExperimentResultsQueryRunner.ts

```diff
@@ -86,6 +95,16 @@ export const startExperimentResultQueries = async (
     segmentObj = await findSegmentById(snapshotSettings.segment, organization);
   }
 
+  const exposureQuery = (integration.settings?.queries?.exposure || []).find(
+    (q) => q.id === snapshotSettings.exposureQueryId
+  );
+  // Add experiment dimensions based on the selected exposure query
+  const availableExperimentDimensions: ExperimentDimension[] = (
```

Comment:
I think this may be where you need to read from the `dimensionsForTraffic` setting instead of `dimensions`

Reply:
Yes, it should be.